### DISCUSSION TITLE
gRPC-Gateway: How to translate error messages

### DISCUSSION THREAD

**Discussion by reckerp** [Kong/kong#13751] [Help]
My gRPC services return error messages, as well as gRPC status codes. Only the status codes get translated. The user of the API does not get any feedback on their actions. Is there a possibility to also translate the messages?

**Comment by ProBrian** [comment]
Hi @reckerp, How does the normal gRPC response from kong works?(if your send correct gRPC request it returns correctly decoded response data from Kong?), and it will be helpful if you can paste the log of kong which is nearby the error occur time.

**Reply by reckerp** [reply]
Hi @ProBrian,
yes Kong does decoding of data properly. It just does not return error messages:

Normal request - Correct result
```
"POST /v1/authentication/login HTTP/1.1" 200 404 "-" "PostmanRuntime/7.42.0" kong_request_id: "ba9e3800af5b326796367526c96c0209" 
```

Incorrect request - Only error code translated, no message
```
"POST /v1/authentication/login HTTP/1.1" 403 0 "-" "PostmanRuntime/7.42.0" kong_request_id: "56be702e27ba8f1173bce05dd606656e"
```

Thank you for your help!

**Comment by ProBrian** [comment]
@reckerp As I understand, gRPC status message are carried by http headers, not body, can you check your response headers, I think you can get error message in headers like: 
```
grpc-status: 3
grpc-message: "some error message"
```
